SourceryRuntime/Sources/Extensions.swift

```diff
@@ -130,47 +135,46 @@ public extension String {
 
     /// :nodoc:
     func components(separatedBy delimiter: String, excludingDelimiterBetween between: (open: String, close: String)) -> [String] {
+        return self.components(separatedBy: delimiter, excludingDelimiterBetween: (between.open.map { String($0) }, between.close.map { String($0) }))
+    }
+
+    /// :nodoc:
+    func components(separatedBy delimiter: String, excludingDelimiterBetween between: (open: [String], close: [String])) -> [String] {
         var boundingCharactersCount: Int = 0
         var quotesCount: Int = 0
         var item = ""
         var items = [String]()
         var matchedDelimiter = (alreadyMatched: "", leftToMatch: delimiter)
 
-        for char in self {
-            if between.open.contains(char) {
-                if !(boundingCharactersCount == 0 && String(char) == delimiter) {
+        var i = self.startIndex
+        while i < self.endIndex {
+            var increaseOffset = 1
+            defer {
+                i = self.index(i, offsetBy: increaseOffset)
+            }
+            let current = self[i..<(self.index(i, offsetBy: delimiter.count, limitedBy: self.endIndex) ?? self.endIndex)]
```

Comment:
what does `current` mean in this context?

Reply:
`current` represents the current "part" of the string being scanned (since it can be more than more character). I was trying to find an equivalence of `char` but for more than one character, but that's the best I could come up with. If you do feel free to say an d I'll update 😉

Reply:
`currentlyScanned` maybe? Definitely longer, but better be clear than short.